packages/cli/src/lib/commands/collect/utils/persist/persist-flow.unit.test.ts

```diff
@@ -0,0 +1,124 @@
+    expect(reports.at(0)).toContain('json');
+  });
+
+  it('should extract an html report from UserFlow if html is given as format', async () => {
+    const generateReportSpy = jest.spyOn(flow, 'generateReport');
+    await persistFlow(flow, { outPath: '', format: ['html'], url: 'mock.com' });
+    expect(generateReportSpy).toHaveBeenCalled();
+  });
+
+  it('should save the report in html if html is given as format', async () => {
+    jest.spyOn(fs, 'existsSync').mockReturnValue(true);
+    jest.spyOn(flow, 'generateReport').mockResolvedValue('Mock HTML Report');
+    jest.spyOn(reportUtils, 'toReportName').mockReturnValue('MockName')
+    await persistFlow(flow, { outPath: 'Flow', format: ['html'], url: 'mock.com' });
+    expect(writeFile).toHaveBeenCalledWith(join('Flow', 'MockName.html'), 'Mock HTML Report');
+  });
+
+  it('should return the path to the html report if html is given as format', async () => {
+    jest.spyOn(fs, 'existsSync').mockReturnValue(true);
+    const reports = await persistFlow(flow, { outPath: '', format: ['html'], url: 'mock.com' });
+    expect(reports.at(0)).toContain('.html');
+  });
+
+  it('should extract an md report from the json report if md is given as format', async () => {
+    jest.spyOn(flow, 'createFlowResult').mockResolvedValue({mock: 'base for md report'});
+    const createReducedReportSpy = jest.spyOn(reportUtils, 'createReducedReport').mockReturnValue({mock: 'reduced report'} as any as ReducedReport);
+    const generateMdReportSpy = jest.spyOn(mdReportUtils, 'generateMdReport');
+    await persistFlow(flow, { outPath: '', format: ['md'], url: 'mock.com' });
+    expect(createReducedReportSpy).toHaveBeenCalledWith({mock: 'base for md report'})
+    expect(generateMdReportSpy).toHaveBeenCalledWith({mock: 'reduced report'});
+  });
+
+  it('should save the report in md if md is given as format', async () => {
+    jest.spyOn(fs, 'existsSync').mockReturnValue(true);
+    jest.spyOn(mdReportUtils, 'generateMdReport').mockReturnValue('Mock Md Report')
+    jest.spyOn(reportUtils, 'toReportName').mockReturnValue('MockName')
+    await persistFlow(flow, { outPath: 'Flow', format: ['md'], url: 'mock.com' });
+    expect(writeFile).toHaveBeenCalledWith(join('Flow', 'MockName.md'), 'Mock Md Report');
+  });
+
+  it('should return the path to the html report if html is given as format', async () => {
+    jest.spyOn(fs, 'existsSync').mockReturnValue(true);
+    const reports = await persistFlow(flow, { outPath: '', format: ['md'], url: 'mock.com' });
+    expect(reports.at(0)).toContain('.md');
+  });
+
+  it('should save the report in multiple formats if multiple formats are given', async () => {
+    jest.spyOn(fs, 'existsSync').mockReturnValue(true);
+    const reports = await persistFlow(flow, { outPath: '', format: ['html', 'json', 'md', 'stdout'], url: 'mock.com' });
+    expect(reports.length).toEqual(3);
```

Comment:
If the `toReportName` mock suggestion above is implemented, one can then assert the whole array.
```suggestion
    expect(reports).toStrictEqual(['report.json', 'report.md', 'report.html']);
```